pkg/3scale/amp/component/s3_options.go

```diff
@@ -2,15 +2,29 @@ package component
 
 import "github.com/go-playground/validator/v10"
 
+const (
+	AwsAccessKeyID          = "AWS_ACCESS_KEY_ID"
+	AwsSecretAccessKey      = "AWS_SECRET_ACCESS_KEY"
+	AwsBucket               = "AWS_BUCKET"
+	AwsRegion               = "AWS_REGION"
+	AwsProtocol             = "AWS_PROTOCOL"
+	AwsHostname             = "AWS_HOSTNAME"
+	AwsPathStyle            = "AWS_PATH_STYLE"
+	AwsRoleArn              = "AWS_ROLE_ARN"
+	AwsWebIdentityTokenFile = "AWS_WEB_IDENTITY_TOKEN_FILE"
+)
+
 type S3Options struct {
-	AwsAccessKeyId       string `validate:"required"`
-	AwsSecretAccessKey   string `validate:"required"`
-	AwsRegion            string `validate:"required"`
-	AwsBucket            string `validate:"required"`
-	AwsProtocol          string `validate:"required"`
-	AwsHostname          string `validate:"required"`
-	AwsPathStyle         string `validate:"required"`
-	AwsCredentialsSecret string `validate:"required"`
+	AwsAccessKeyId          string `validate:"required"`
+	AwsSecretAccessKey      string `validate:"required"`
+	AwsRoleArn              string `validate:"required"`
+	AwsWebIdentityTokenFile string `validate:"required"`
+	AwsRegion               string `validate:"required"`
+	AwsBucket               string `validate:"required"`
+	AwsProtocol             string `validate:"required"`
+	AwsHostname             string `validate:"required"`
+	AwsPathStyle            string `validate:"required"`
+	AwsCredentialsSecret    string `validate:"required"`
 }
 
 func NewS3Options() *S3Options {
```

Comment:
this func can be removed

Reply:
file s3_options.go - removed, consts moved to system_options.go